test/InitControlTest.cpp

```diff
@@ -0,0 +1,141 @@
+    EXPECT_CALL(m_platform_io,
+                write_control("FAKE_CONTROL5", PlatformTopo::domain_name_to_type("package"), 3, 1e9));
+
+    m_init_control = std::make_shared<InitControlImp>(m_file_name, m_platform_io);
+    m_init_control->write_controls();
+}
+
+TEST_F(InitControlTest, throw_bad_input)
+{
+    std::string contents = "CPU_POWER_LIMIT package 0\n";
+    WriteFile(contents);
+    GEOPM_EXPECT_THROW_MESSAGE(std::make_shared<InitControlImp>(m_file_name, m_platform_io),
+                               GEOPM_ERROR_INVALID, "missing fields");
+
+    contents = "CPU_POWER_LIMIT package 0 2 00\n";
+    WriteFile(contents);
+    GEOPM_EXPECT_THROW_MESSAGE(std::make_shared<InitControlImp>(m_file_name, m_platform_io),
+                               GEOPM_ERROR_INVALID, "Data found in comment field");
+
+    contents = "CPU_POWER_LIMIT package 0 2#00\n";
+    WriteFile(contents);
+    GEOPM_EXPECT_THROW_MESSAGE(std::make_shared<InitControlImp>(m_file_name, m_platform_io),
+                               GEOPM_ERROR_INVALID, "bad input: #00");
+
+    contents = "CPU_POWER_LIMIT package 0 seven\n";
+    WriteFile(contents);
+    GEOPM_EXPECT_THROW_MESSAGE(std::make_shared<InitControlImp>(m_file_name, m_platform_io),
+                               GEOPM_ERROR_INVALID, "Missing setting value");
+
+    contents = "CPU_POWER_LIMIT package 0 200 CPU_POWER_LIMIT package 0 150";
+    WriteFile(contents);
+    GEOPM_EXPECT_THROW_MESSAGE(std::make_shared<InitControlImp>(m_file_name, m_platform_io),
+                               GEOPM_ERROR_INVALID, "Data found in comment field");
+
+    contents = "CPU_POWER_LIMIT package one -7.77\n";
+    WriteFile(contents);
+    GEOPM_EXPECT_THROW_MESSAGE(std::make_shared<InitControlImp>(m_file_name, m_platform_io),
+                               GEOPM_ERROR_INVALID, "missing fields");
+
+    contents = "CPU_POWER_LIMIT 1 1 -7.77\n";
+    WriteFile(contents);
+    GEOPM_EXPECT_THROW_MESSAGE(std::make_shared<InitControlImp>(m_file_name, m_platform_io),
+                               GEOPM_ERROR_INVALID, "unrecognized domain_name");
+}
+
+TEST_F(InitControlTest, throw_invalid_write)
+{
+    std::string contents = "FAKE_CONTROL0 board 0 123\n"
+                           "FAKE_CONTROL1 package 1 -7.77\n"
+                           "FAKE_CONTROL2 gpu 3 0\n";
+    WriteFile(contents);
+
+    InSequence s;
+    EXPECT_CALL(m_platform_io,
+                write_control("FAKE_CONTROL0", PlatformTopo::domain_name_to_type("board"), 0, 123));
+    EXPECT_CALL(m_platform_io,
+                write_control("FAKE_CONTROL1", PlatformTopo::domain_name_to_type("package"), 1, -7.77))
+        .WillOnce(Throw(geopm::Exception("Test-injected failure", GEOPM_ERROR_INVALID, __FILE__, __LINE__)));
+
+    m_init_control = std::make_shared<InitControlImp>(m_file_name, m_platform_io);
+    GEOPM_EXPECT_THROW_MESSAGE(m_init_control->write_controls(),
+                               GEOPM_ERROR_INVALID, "Test-injected failure");
+}
```

Comment:
Add a test that specifies the expected behavior when an empty file name is provided to the constructor